File: src/main/java/com/sfh/pokeRogueBot/phase/impl/SelectGenderPhase.java
package com.sfh.pokeRogueBot.phase.impl;

import com.sfh.pokeRogueBot.model.browser.enums.GameMode;
import com.sfh.pokeRogueBot.model.exception.NotSupportedException;
import com.sfh.pokeRogueBot.phase.AbstractPhase;
import com.sfh.pokeRogueBot.phase.Phase;
import com.sfh.pokeRogueBot.phase.actions.PhaseAction;
import org.springframework.stereotype.Component;

@Component
public class SelectGenderPhase extends AbstractPhase implements Phase {

    @Override
    public String getPhaseName() {
        return Phase.SELECT_GENDER_PHASE;
    }

    @Override
    public PhaseAction[] getActionsForGameMode(GameMode gameMode) throws NotSupportedException {
        return new PhaseAction[]{
                this.pressSpace
        };
    }
}

File: src/main/kotlin/com/sfh/pokeRogueBot/phase/impl/SelectGenderPhase.kt
package com.sfh.pokeRogueBot.phase.impl

import com.sfh.pokeRogueBot.model.enums.UiMode
import com.sfh.pokeRogueBot.model.exception.NotSupportedException
import com.sfh.pokeRogueBot.phase.UiPhase
import com.sfh.pokeRogueBot.service.javascript.JsUiService
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

@Component
class SelectGenderPhase(
    @Value("\${app.game.chooseGender}") private val genderToPick: String,
    private val jsUiService: JsUiService,
) : UiPhase {
    override val phaseName = "SelectGenderPhase"

    companion object {
        val BOY = "BOY"
    }

    @Throws(NotSupportedException::class)
    override fun handleUiMode(uiMode: UiMode) {
        val indexToSetCursorTo = if (BOY.equals(genderToPick)) 0 else 1
        jsUiService.setUiHandlerCursor(uiMode, indexToSetCursorTo)
        jsUiService.sendActionButton()
    }
}